test/Interop/Cxx/operators/member-inline.swift

```diff
@@ -90,4 +90,40 @@ OperatorsTestSuite.test("DifferentTypesArray.subscript (inline)") {
   expectEqual(1.5.rounded(.up), resultDouble.rounded(.up))
 }
 
+OperatorsTestSuite.test("ReadOnlyIntArrayByVal.subscript (inline)") {
+  var arr = ReadOnlyIntArrayByVal(1)
+
+  let result0 = arr[0]
+  let result2 = arr[2]
+  let result4 = arr[4]
+
+  expectEqual(1, result0)
+  expectEqual(3, result2)
+  expectEqual(5, result4)
+
+  arr.setValueAtIndex(42, 2)
+  let result5 = arr[2]
+  expectEqual(42, result5)
+}
+
+OperatorsTestSuite.test("DifferentTypesArrayByVal.subscript (inline)") {
+  var arr = DifferentTypesArrayByVal()
+
+  let resultInt: Int32 = arr[2]
+  let resultDouble: Double = arr[0.1]
+
+  expectEqual(3, resultInt)
+  expectEqual(1.5.rounded(.down), resultDouble.rounded(.down))
+  expectEqual(1.5.rounded(.up), resultDouble.rounded(.up))
+}
+
+OperatorsTestSuite.test("ReadOnlyNonTrivialArrayByVal.subscript (inline)") {
+  var arr = ReadOnlyNonTrivialArrayByVal()
+  if let CStr = arr[0].Str {
```

Comment:
@plotfi  this approach looks good to me! With the optionals I think you might need to do either 
1. adjust the logic that imports `__operatorSubscriptConst` so that its return type is `UnsafePointer<Int32>?` instead of `UnsafePointer<Int32>!` or
2. add some kind of conversion between optional & an implicitly-unwrapped optional in the `get`/`set` accessor synthesis logic (`synthesizeSubscriptGetterBody`/`synthesizeSubscriptSetterBody`).